### PR TITLE
fix: Embed Code Generator - Fixes 404 in preview of embed for orgs

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?
Demo https://www.loom.com/share/6513fbd2cda442a589306c46e676b525

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)

## Tests
-  See the loom

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist
- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: apps/web/components/Embed.tsx [hunk 1]
```diff
@@ -323,18 +344,22 @@ ${uiInstructionCode}`;
 ${uiInstructionCode}`;
     },
   },
-};
+} satisfies Record<string, Record<string, (...args: any[]) => string>>;
```
Comment (hariombalhara) [inline apps/web/components/Embed.tsx:347]: satisfies does what I really needed. It works now with Next.js

File: apps/web/components/Embed.tsx [hunk 2]
```diff
@@ -687,7 +730,14 @@ ${getEmbedTypeSpecificString({ embedFramework: "HTML", embedType, calLink, previ
 
 /* If you are using npm */
 // npm install @calcom/embed-react
-${getEmbedTypeSpecificString({ embedFramework: "react", embedType, calLink, previewState })}
+${getEmbedTypeSpecificString({
```
Comment (hariombalhara) [inline apps/web/components/Embed.tsx:733]: Thought of making it a hook. But embedFramework is conditional and using a hook with conditional would require me to make a state which is unnecessary. So, keeping it as a regular function only

File: packages/embeds/embed-core/src/preview.ts [hunk 3]
```diff
@@ -1,7 +1,11 @@
-const WEBAPP_URL =
-  import.meta.env.EMBED_PUBLIC_WEBAPP_URL || `https://${import.meta.env.EMBED_PUBLIC_VERCEL_URL}`;
-const EMBED_LIB_URL = import.meta.env.EMBED_PUBLIC_EMBED_LIB_URL || `${WEBAPP_URL}/embed/embed.js`;
-
```
Comment (hariombalhara) [inline packages/embeds/embed-core/src/preview.ts:4]: Orgs change forced this logic to improve. No need to read from env variables here.

**Review by zomars — DISMISSED** [review]
[cool](https://app.graphite.dev/macros/calcom)

**Review by zomars — APPROVED** [review]
Fixed conflicts

**Pull request merged**